lms/djangoapps/courseware/views/views.py

```diff
@@ -1060,6 +1060,16 @@ def dates(request, course_id):
         )
         request.user = masquerade_user
 
+    user_access = {
+        'is_enrolled': CourseEnrollment.is_enrolled(request.user, course_key),
+        'is_staff': has_access(request.user, 'staff', course_key),
+    }
+
+    # Render the full content to enrolled users, as well as to course and global staff.
+    # Unenrolled users who are not course or global staff are redirected to the Outline Tab.
```

Comment:
Global staff? That implies to me Django staff (user.is_staff), vs users with staff access to a course (like instructors, TAs, etc). This code is currently checking for course-staff access. Which is the intent here?

Reply:
(Our naming around all this sucks.)

Reply:
Oh sorry, I didn't read that this comment keeps saying "course or global" -- duh. I just saw the global bit. So ignore my comment above.